jetbrains-ultimate/tst/software/aws/toolkits/jetbrains/services/DevEnvStatusWatcherTest.kt

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services
+
+import com.intellij.openapi.ui.TestDialog
+import com.intellij.openapi.ui.TestDialogManager
+import com.intellij.testFramework.ProjectRule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import software.aws.toolkits.jetbrains.services.caws.DevEnvStatusWatcher
+
+class DevEnvStatusWatcherTest {
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    @Test
+    fun `Heartbeat check stops if no response is returned by the API`() {
+        val sut = DevEnvStatusWatcher()
+        val devEnvStatusWatcher = spy<DevEnvStatusWatcher>(sut) {
+            doReturn(600.toLong()).whenever(it).getJbRecordedActivity()
+            doReturn(null).whenever(it).getLastRecordedApiActivity()
+        }
+        val response = devEnvStatusWatcher.checkHeartbeat(0, 0, projectRule.project)
+        assertThat(response.first).isTrue()
+    }
+
+    @Test
+    fun `API is called if user extends the timeout 5 minutes before inactivity timeout`() {
```

Comment:
API is not called if user rejects?

Reply:
No, it isn't cause at that point we dont want to update the API. The API at this point will have the last recorded activity. Since the user doesn't want to continue. the session if they reject it, we needn't update the timestamp.

Reply:
Added a test for the condition "if the user rejects"